components/Balances.tsx

```diff
@@ -7,14 +7,9 @@ export function Balances() {
   const { data: balances } = useBalances();
 
   return balances
-    ? balances
-        .map((bal) => bal.balanceView)
-        .map((balanceView) => (
-          <BalanceRow
-            key={balanceView!.toJsonString()}
-            balance={balanceView!}
-          />
-        ))
+    ? balances.map(({ balanceView }) => (
+        <BalanceRow key={balanceView!.toJsonString()} balance={balanceView!} />
```

Comment:
suggestion: generally it's better to just use a conditional instead of a `!`. Because if something goes wrong, you'll get a readable error versus something like `could not find toJsonString of undefined`. It would look something like:

```ts
if (!balanceView) throw new Error("No balance view on balance data")
```

Reply:
agree, added an additional check: 
```  return balances && balances.every((b) => b.balanceView !== undefined)```